File: Tutorial3-1LiveData-DataBinding/src/main/java/com/example/tutorial3livedataanddatabinding/MainActivity.java
package com.example.tutorial3livedataanddatabinding;

import androidx.lifecycle.ViewModelProviders;
import androidx.databinding.DataBindingUtil;
import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;

import com.example.tutorial3livedataanddatabinding.databinding.ActivityMainBinding;
import com.example.tutorial3livedataanddatabinding.viewmodel.CounterViewModel;

public class MainActivity extends AppCompatActivity {


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ActivityMainBinding binding = DataBindingUtil.setContentView(this, R.layout.activity_main);

        /*
            LifeCycleOwner should be set for LiveData changes to be propagated to UI for this binding
         */
        binding.setLifecycleOwner(this);

        CounterViewModel counterViewModel = ViewModelProviders.of(this).get(CounterViewModel.class);

        binding.setViewModel(counterViewModel);

    }

}

File: Tutorial3-1LiveData-DataBinding/src/main/java/com/example/tutorial3livedataanddatabinding/MainActivity.kt
package com.example.tutorial3livedataanddatabinding

import android.os.Bundle
import androidx.activity.viewModels
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import com.example.tutorial3livedataanddatabinding.databinding.ActivityMainBinding
import com.example.tutorial3livedataanddatabinding.viewmodel.CounterViewModel

class MainActivity : AppCompatActivity() {

    private val counterViewModel: CounterViewModel by viewModels()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val binding: ActivityMainBinding = DataBindingUtil.setContentView(this, R.layout.activity_main)

        // 🔥 ‍LifeCycleOwner should be set for LiveData changes to be propagated to UI for this binding
        binding.lifecycleOwner = this

        binding.viewModel = counterViewModel
    }
}

/*
If you use Java, do the following :

ActivityMainBinding binding = DataBindingUtil.setContentView(this, R.layout.activity_main);
binding.setLifecycleOwner(this);
CounterViewModel counterViewModel = ViewModelProviders.of(this).get(CounterViewModel.class);
binding.setViewModel(counterViewModel);

* */